### ISSUE TITLE
node-server Daemonset: unable to execute command" err="DirectPVNode.directpv.min.io

### ISSUE THREAD

**Issue by naguam** [minio/directpv#959]
**Describe the bug**
node-server Daemonset in CrashloopBackOff on a clean install on a kind cluster.

**To Reproduce**
Clean install on a kind cluster with kubernetes 1.31.

```sh
#!/usr/bin/bash

mkdir /scratch/kind-control-plane
mkdir /scratch/kind-worker
mkdir /scratch/kind-worker2

kind create cluster --config multinode.yml
```

```yaml
---
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  extraMounts:
  - hostPath: /scratch/kind-control-plane
    containerPath: /var/local-path-provisioner
- role: worker
  extraMounts:
  - hostPath: /scratch/kind-worker
    containerPath: /var/local-path-provisioner
- role: worker
  extraMounts:
  - hostPath: /scratch/kind-worker2
    containerPath: /var/local-path-provisioner
...
```

`kubectl directpv install`

**Expected behavior**
node-server should not be in CrashloopBackOff nor Error.

**Screenshots and logs**

```
  node-controller:
    Container ID:  containerd://dc6b480f2a16ab920c28393f546d4d2bb53600e0016c11ead29ad4d3677eb64a
    Image:         quay.io/minio/directpv:v4.1.4
    Image ID:      quay.io/minio/directpv@sha256:83fd05fe114ed15c3975333c90cbe18c782d9c4d5c7ad6fdb8cc835e380ba505
    Port:          <none>
    Host Port:     <none>
    Args:
      node-controller
      -v=3
      --kube-node-name=$(KUBE_NODE_NAME)
    State:      Terminated
      Reason:   Error
      Message:  E1010 09:06:13.277883   66349 main.go:148] "unable to execute command" err="DirectPVNode.directpv.min.io \"kind-worker\" is invalid: status.devices: Required value"

      Exit Code:  1
      Started:    Thu, 10 Oct 2024 10:06:13 +0100
      Finished:   Thu, 10 Oct 2024 10:06:13 +0100
    Last State:   Terminated
      Reason:     Error
      Message:    E1010 09:05:56.773748   66257 main.go:148] "unable to execute command" err="DirectPVNode.directpv.min.io \"kind-worker\" is invalid: status.devices: Required value"

      Exit Code:    1
      Started:      Thu, 10 Oct 2024 10:05:56 +0100
      Finished:     Thu, 10 Oct 2024 10:05:56 +0100
    Ready:          False
    Restart Count:  2
```
Note1: kind-worker is my worker node name.

Note2: I configured kind so that the the docker volume mounted into the node for kubernetes local persistent storage is on an XFS drive.

If applicable, add screenshots to help explain your problem. Or logs from the pods in directpv-min-io namespace.

**Deployment information (please complete the following information):**
 - DirectPV version: v4.1.4
 - Kubernetes Version: 1.31
 - OS info: Ubuntu 22.04.5 (But kind node images are Debian 12)
- Kernel version: 6.8.0-45 Ubuntu

**Additional Note**
The doc says the daemonset needs Port 30443 for readiness handlers
It doesn't say in which dimension.
Should I create a Loadbalancer SVC ?
Not sure it is related to the error though.

**Comment by balamurugana** [comment]
@naguam We do not support `Kind` as it doesn't suit for DirectPV workload. You can try `minikube` or some other variant. If you still want to try check `/run/udev/data` is exported into `Kind` and check your kubelet directory and use installation accordingly.

**Comment by balamurugana** [comment]
> Hi,
> 
> /run/udev/data exists in the containers but are empty.
> 

You have to export it to containers.

> I haven't fully understood.
> 
> Is directpv meant to use direct raw block storage devices ?
> 

Yes.

> I was using kind for learning and also to prototype a cluster for future production. (Experimenting with ArgoCD operator and ArgoCD, Minio Operator, and potentially Minio with an end goal to have minio for argo workflow)
> 

For DirectPV, `Kind` is not the right choice.

> I must say i'm not so keen of installing yet another kubernetes distribution on my computer. Do you think that the localpath provisioning is enough then for minio ?
> 

As I told you we do not support `Kind` for DirectPV. It is your call.

> Or an external CSI (NFS with subdirs or iscsi) ?
> 

Check https://github.com/minio/directpv/blob/master/docs/faq.md

> Maybe I should also try mount propagation https://kind.sigs.k8s.io/docs/user/configuration/#extra-mounts.
> 
> Cheers.

**Comment by naguam** [comment]
Thanks for the answer.

> > Is directpv meant to use direct raw block storage devices ?
> Yes.

Thanks for that, I don't know how I did not understand that despite reading the installation doc.
Maybe the telling as an alternative to Local Persistent Volumes induced me to the misunderstanding.

> For DirectPV, Kind is not the right choice.

Yes I already understood from your first answer.
I was looking for alternatives viable enough for Minio for testing purposes.

> As I told you we do not support Kind for DirectPV. It is your call.

I did not contradict you on that.
